### PR TITLE
Feat/logs viewer

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: templates/pages/admin/log_viewer.html.twig [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                    <a href="?action=download_log_file&amp;filename={{ filename }}"
+                    class="btn btn-sm btn-outline-secondary"
+                    title="{{ __("Download file") }}"
+                    data-bs-toggle="tooltip" data-bs-placement="bottom">
+                        <i class="ti ti-device-floppy"></i>
+                    </a>
+                </div>
+
+                <span class="input-icon ms-3">
+                    <input type="text" value="" class="form-control form-control-sm" id="filter_logs_input"
```
Comment (cconard96) [inline templates/pages/admin/log_viewer.html.twig:87]:
```suggestion
                    <input type="text" value="" class="form-control form-control-sm" id="filter-logs-input"
```

File: templates/pages/admin/log_viewer.html.twig [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                        placeholder="{{ __("filter logs") }}">
+                    <span class="input-icon-addon">
+                        <i class="ti ti-filter fa-xs"></i>
+                    </span>
+                </span>
+            </span>
+            <span class="d-inline-flex">
+                {% set limitdropdown = include('components/dropdown/limit.html.twig', {
```
Comment (cconard96) [inline templates/pages/admin/log_viewer.html.twig:95]: This dropdown currently redirects to an invalid page `/front/glpilist_limit=5` when changed. No `href` is set and it isn't in a tab, so the default behavior doesn't work. Also, I'm not sure the fluid_search check is needed as the page doesn't initialize a fluid search view.

File: src/LogViewer.php [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+     * @return void
+     */
+    public static function showLogFile(string $fileslug = "", bool $only_content = false, int $nb_lines = null)
+    {
+        global $CFG_GLPI;
+
+        $logfiles = self::getLogsFilesList();
+        $filename = $logfiles[$fileslug] ?? "";
+        $filepath = GLPI_LOG_DIR . "/" . $filename;
+        if (!file_exists($filepath)) {
```
Comment (cconard96) [inline src/LogViewer.php:131]:
```suggestion
        if (is_dir($filepath) || !file_exists($filepath)) {
```
An invalid log name would make this code try reading from the log directory.

**Review by cconard96 — CHANGES_REQUESTED** [review]

**Comment by orthagh** [comment]
> Some remarks about the UX:

all should be fixed (or get around)

**Comment by orthagh** [comment]
You're right, it should be fixed

**Review by cconard96 — APPROVED** [review]

**Comment by trasher** [comment]
Maybe some tests may be added on new  LogViewer class

**Comment by orthagh** [comment]
> Maybe some tests may be added on new LogViewer class

Done, and also class has been "un-staticked" (as much as possible)

**Comment by trasher** [comment]
Seems correct :+1: 

Would it be possible to add the last modification date in logs files view? And maybe order files by that date, so non accessed files are not displayed first.

**Comment by trasher** [comment]
Maybe log file size may be of interest also; and why not the ability for super admins to empty them?

File: templates/pages/admin/logs_list.html.twig [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			<div class="card">
+				<div class="card-header justify-content-between">
+					<h3>{{ __("Log files") }}</h3>
+
+					<div class="d-inline-flex">
+						{% set sort_controls = [
+                            {
+                                'key': 'filename',
+                                'icon': 'ti ti-alphabet-latin',
+                                'label': __("sort by filename")
```
Comment (cconard96) [inline templates/pages/admin/logs_list.html.twig:60]:
```suggestion
                                'label': __("Sort by filename")
```

File: templates/pages/admin/logs_list.html.twig [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                            }, {
+                                'key': 'size',
+                                'icon': 'ti ti-file',
+                                'label': __("sort by size")
```
Comment (cconard96) [inline templates/pages/admin/logs_list.html.twig:64]:
```suggestion
                                'label': __("Sort by size")
```

File: templates/pages/admin/logs_list.html.twig [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                            }, {
+                                'key': 'datemod',
+                                'icon': 'ti ti-clock',
+                                'label': __("sort by date")
```
Comment (cconard96) [inline templates/pages/admin/logs_list.html.twig:68]:
```suggestion
                                'label': __("Sort by date")
```

**Review by cconard96 — CHANGES_REQUESTED** [review]

**Review by trasher — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**